OpenRA.Mods.RA2/Traits/Garrisoner.cs

```diff
@@ -0,0 +1,191 @@
+        public readonly int Weight = 1;
+
+        [Desc("Use to set when to use alternate transports (Never, Force, Default, Always).",
+            "Force - use force move modifier (Alt) to enable.",
+            "Default - use force move modifier (Alt) to disable.")]
+        public readonly AlternateTransportsMode AlternateTransportsMode = AlternateTransportsMode.Default;
+
+        [Desc("Number of retries using alternate transports.")]
+        public readonly int MaxAlternateTransportAttempts = 1;
+
+        [Desc("Range from self for looking for an alternate transport (default: 5.5 cells).")]
+        public readonly WDist AlternateTransportScanRange = WDist.FromCells(11) / 2;
+
+        [Desc("Valid Stances. Default Ally")]
+        public readonly Stance TargetStances = Stance.Ally;
+
+        [GrantedConditionReference]
+        [Desc("The condition to grant to when this actor is loaded inside any transport.")]
+        public readonly string GarrisonCondition = null;
+
+        [Desc("Conditions to grant when this actor is loaded inside specified transport.",
+            "A dictionary of [actor id]: [condition].")]
+        public readonly Dictionary<string, string> GarrisonConditions = new Dictionary<string, string>();
+
+        [GrantedConditionReference]
+        public IEnumerable<string> LinterGarrisonConditions { get { return GarrisonConditions.Values; } }
+
+        [VoiceReference] public readonly string Voice = "Action";
+
+        public object Create(ActorInitializer init) { return new Garrisoner(this); }
+    }
+
+    public class Garrisoner : INotifyCreated, IIssueOrder, IResolveOrder, IOrderVoice, INotifyRemovedFromWorld, INotifyEnteredGarrison, INotifyExitedGarrison
+    {
+        public readonly GarrisonerInfo Info;
+        public Actor Transport;
+
+        ConditionManager conditionManager;
+        int anyGarrisonToken = ConditionManager.InvalidConditionToken;
+        int specificGarrisonToken = ConditionManager.InvalidConditionToken;
+
+        public Garrisoner(GarrisonerInfo info)
+        {
+            Info = info;
+            Func<Actor, Actor, bool> canTarget = IsCorrectGarrisonType;
+            Func<Actor, Actor, bool> useEnterCursor = CanEnter;
+            Orders = new EnterGarrisonTargeter<GarrisonInfo>[]
+            {
+                new EnterGarrisonTargeter<GarrisonInfo>("EnterTransport", 5, canTarget, useEnterCursor, Info.AlternateTransportsMode)
+            };
+        }
+
+        public Garrison ReservedGarrison { get; private set; }
+
+        public IEnumerable<IOrderTargeter> Orders { get; private set; }
+
+        void INotifyCreated.Created(Actor self)
+        {
+            conditionManager = self.TraitOrDefault<ConditionManager>();
+        }
+
+        public Order IssueOrder(Actor self, IOrderTargeter order, Target target, bool queued)
+        {
+            if (order.OrderID == "EnterTransport" || order.OrderID == "EnterTransports")
```

Comment:
I don't see the alternative transports logic reimplemented so you can drop the checks for them.

Although I'm not sure if npt implementing the alternative garrisoning is a good or bad thing. It's alt-click at transports.

Reply:
I would like to see alternative garrisoning being implemented, would make it easier to garrison across mulipile structures.

Reply:
If we go that waya, we definitely need the orderstring changes to don't mix plain transporting and garrisoning.

Reply:
Done.